libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -79,62 +87,67 @@ export class TimeMapComponent implements OnInit {
+        const selectedCounty = this.countyData.filter((county) => county.fips === props.fips);
         this.infoBoxModel.next({
-          name: feature.properties.NAME,
-          pop: feature.properties.POPESTIMATE2019,
-          cases: selectedCounty[0]['confirmed'],
-          infectionRate: selectedCounty[0]['infection_rate'],
-          deaths: selectedCounty[0]['deaths'],
-          deathRate: selectedCounty[0]['death_rate']
+          name: props.NAME,
+          population: selectedCounty[0].population,
+          confirmed: selectedCounty[0].confirmed,
+          infection_rate: selectedCounty[0].infection_rate,
+          deaths: selectedCounty[0].deaths,
+          death_rate: selectedCounty[0].death_rate
         });
       });
 
       map.on('mousemove', 'covid-state', (e) => {
-        const feature = e.features[0];
-        const selectedState = this.stateData.filter((state) => state.STATE === feature.properties.STATE);
-
+        const [feature] = e.features;
+        const props = feature.properties as State;
+        const selectedState = this.stateData.filter((state) => state.STATE === props.STATE);
         this.infoBoxModel.next({
           name: feature.properties.NAME,
-          pop: feature.properties.POPESTIMATE2019,
-          cases: selectedState[0]['confirmed'],
-          infectionRate: selectedState[0]['infection_rate'],
-          deaths: selectedState[0]['deaths'],
-          deathRate: selectedState[0]['death_rate']
+          population: selectedState[0].population,
+          confirmed: selectedState[0].confirmed,
+          infection_rate: selectedState[0].infection_rate,
+          deaths: selectedState[0].deaths,
+          death_rate: selectedState[0].death_rate
         });
       });
+
+      map.on('mouseleave', 'covid-county', (e) => {
+        this.infoBoxModel.next(null);
+      });
+
+      map.on('mouseleave', 'covid-state', (e) => {
+        this.infoBoxModel.next(null);
+      });
+
+      map.dragRotate.disable();
+
       this.reloadData();
     });
   }
 
-  public reloadData() {
-    const currentDateSelected = '2020-06-06';
-    console.log(currentDateSelected);
-    const stateURL =
-      'https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/states/' +
-      currentDateSelected +
-      '.json';
-    const countyURL =
-      'https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/counties/' +
-      currentDateSelected +
-      '.json';
+  public reloadData(): void {
+    this.dateSelected = this.datePicker.nativeElement.value;
+    const stateURL = `https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/states/${this.dateSelected}.json`;
+    const countyURL = `https://raw.githubusercontent.com/jorge-sepulveda/covid-time-map/master/src/pyscraper/outputFiles/counties/${this.dateSelected}.json`;
+    this.dateModel.next(this.dateSelected);
```

Comment:
Refactor. Let the model be the source of truth, and not the view input value.

Reply:
switched and removed DatePicker as there is no use for it anymore